### DISCUSSION TITLE
When the Buffer memory will release?

### DISCUSSION THREAD

**Discussion by sseanwang** [gpuweb/gpuweb#3330] [Q&A]
```for (let i = 0; i < 500; i++) {
      let buffer = device.createBuffer({
        size: 6553600,
        usage: GPUBufferUsage.VERTEX,
        mappedAtCreation: true,
      });
      buffer.unmap();
      buffer.destroy();
    }
```
I'm using this code to test the buffer memory creation and release, will the buffer.destroy suppose to release all the memory used by this buffer? But I didn't see any memory released in my windows 11 machine. So how can I free the buffer memory?

**Comment by Kangz** [comment]
`buffer.destroy()` tells the WebGPU implementation that this buffer can be freed after all current operations using the buffer are done being used. When the buffer itself is freed is unobservable. For example, at the moment, in Chromium, operations are not sent to be executed by the "GPU process" until they are flushed, and here there are no operations that would flush the command stream, so the data isn't freed. Operations that cause flushes are anything returning a promise, queue submission, and maybe some others.

**Reply by kainino0x** [reply]
Also note, the memory of the buffer is GPU memory, not CPU memory. So if you are watching a system monitor to see if the buffer got destroyed, make sure you are looking at VRAM usage.
